Infrastructure/Services/ParcelService.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Infrastructure.Interfaces;
+using Infrastructure.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.DataAccess
+{
+    public class ParcelService : IParcelService
+    {
+        private readonly DatabaseContext _dbContext;
+
+        public ParcelService(DatabaseContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<Parcel> GetByTrackingId(string trackingId)
+        {
+            try
+            {
+                var parcel = await _dbContext.Parcels.FirstOrDefaultAsync(x => x.TrackingNumber == trackingId);
+                if (parcel == null)
+                {
+                    return null;
+                }
+                return parcel;
+            }
+            catch
+            {
+                throw;
+            }
```

Comment:
I can see a lot of places where inside catch, you simply rethrow. Then what's the point of try/catch as without it it would just throw error further without rethrowing?